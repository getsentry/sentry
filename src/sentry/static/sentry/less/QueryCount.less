.query-count {
  margin-left: 4px;

  &.inline {
    display: inline-block;
  }
}
