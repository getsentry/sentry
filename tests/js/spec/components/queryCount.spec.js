import React from 'react';
import {shallow} from 'sentry-test/enzyme';
import QueryCount from 'app/components/queryCount';

describe('QueryCount', function() {
  it('displays count when no max', function() {
    const wrapper = shallow(<QueryCount count={5} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('displays count when count < max', function() {
    const wrapper = shallow(<QueryCount count={5} max={500} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render if count is 0', function() {
    const wrapper = shallow(<QueryCount count={0} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('can render when count is 0 when `hideIfEmpty` is false', function() {
    const wrapper = shallow(<QueryCount count={0} hideIfEmpty={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays max count if count >= max', function() {
    const wrapper = shallow(<QueryCount count={500} max={500} />);
    expect(wrapper).toMatchSnapshot();
  });
});
