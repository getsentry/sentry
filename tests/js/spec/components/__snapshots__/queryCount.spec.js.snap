// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryCount can render when count is 0 when \`hideIfEmpty\` is false 1`] = `
<QueryCount
  count={0}
  hideIfEmpty={false}
>
  <div
    className="query-count inline"
  >
    <span>
      (
    </span>
    <span
      className="query-count-value"
    >
      0
    </span>
    <span>
      )
    </span>
  </div>
</QueryCount>
`;

exports[`QueryCount displays count when count < max 1`] = `
<QueryCount
  count={5}
  max={500}
>
  <div
    className="query-count inline"
  >
    <span>
      (
    </span>
    <span
      className="query-count-value"
    >
      5
    </span>
    <span>
      )
    </span>
  </div>
</QueryCount>
`;

exports[`QueryCount displays count when no max 1`] = `
<QueryCount
  count={5}
>
  <div
    className="query-count inline"
  >
    <span>
      (
    </span>
    <span
      className="query-count-value"
    >
      5
    </span>
    <span>
      )
    </span>
  </div>
</QueryCount>
`;

exports[`QueryCount displays max count if count >= max 1`] = `
<QueryCount
  count={500}
  max={500}
>
  <div
    className="query-count inline"
  >
    <span>
      (
    </span>
    <span
      className="query-count-value"
    >
      500+
    </span>
    <span>
      )
    </span>
  </div>
</QueryCount>
`;

exports[`QueryCount does not render if count is 0 1`] = `
<QueryCount
  count={0}
/>
`;
